### ISSUE TITLE
Define `graphql` as peer dependency

### ISSUE THREAD

**Issue by mattiloh** [geek/graphi#36]
Thanks for this neat module @geek! Would you be open for a PR that converts `graphql` to a peer dependency? We're experiencing problems while we try to setup the [GraphQL Code Generator ](https://the-guild.dev/graphql/codegen) in a repo that uses `graphi`. We get this message when using the GraphQL endpoint:
``` 
Duplicate "graphql" modules cannot be used at the same time since different
versions may have different capabilities and behavior.
```
`npm ls graphql` shows that `graphi` is set to `graphql@16.6.0` (coming from `16.6.x` in graphi's package.json), while all other packages use `graphql@16.8.1`.

I guess switching the dependency type would result in a breaking change, but it would definitely make sense in terms of interoperability with other modules.

**Comment by geek** [comment]
@mattiloh what is your opinion on 
1. Updating the graphql dep to match 16.8.1 then release graphi as v9.1.1  
2. Introducing the breaking change to make graphql a peer dep and releasing graphi as v10.0.0

**Comment by mattiloh** [comment]
Thanks for your quick response @geek! I believe that option 1 (the patch release) should also resolve the problem effectively. My recommendation is to set the version as `16.x.x` to ensure greater compatibility.

Upon further reflection, I realize that converting it to a peer dependency may not significantly impact the outcome, especially since npm7+ automatically installs peer dependencies AFAIK. Therefore, I agree: option 1 seems like a good choice.

Should I create a PR?

**Comment by geek** [comment]
@mattiloh sounds good. Please updated to 16.x.x as you suggest in a PR and I can merge/release.

**Comment by geek** [comment]
Published the latest as [v9.1.1](https://www.npmjs.com/package/graphi)
